server/app/views/components/QuestionBank.java

```diff
@@ -68,32 +70,32 @@ public QuestionBank setQuestions(ImmutableList<QuestionDefinition> questionDefin
     return this;
   }
 
-  public ContainerTag getContainer() {
+  public FormTag getContainer() {
     return questionBankPanel();
   }
 
-  private ContainerTag questionBankPanel() {
-    ContainerTag questionForm =
-        form(this.csrfTag).withMethod(HttpVerbs.POST).withAction(questionAction);
+  private FormTag questionBankPanel() throws RuntimeException {
+    if (this.maybeCsrfTag.isEmpty()) {
+      throw new RuntimeException("trying to access empty csrf tag for rendering");
```

Comment:
I'd consider not throwing an exception here and instead just letting the user request fail downstream.

Reply:
should I do the same [here](https://github.com/syntapy/universal-application-tool/blob/31e95063d292f4e0bbdf024eb8d5ce3ff4424b4e/server/app/views/components/LinkElement.java#L123-L129) too?